inbox/crispin.py

```diff
@@ -832,7 +832,10 @@ def search_uids(self, criteria: List[str]) -> List[int]:
         criteria.
 
         """
-        return sorted(int(uid) for uid in self.conn.search(criteria))
+        return sorted(
+            int(uid) if not isinstance(uid, int) else uid
+            for uid in self.conn.search(criteria)
```

Comment:
Those are most of the time ints already (if the server follows IMAP spec), there are some nonconforming servers (minority) where we need to cast to int, but let's not do useless work on millions of them.